botorch/utils/sampling.py

```diff
@@ -154,22 +158,30 @@ def draw_sobol_samples(
             to lower and upper bounds, respectively.
         n: The number of (q-batch) samples.
         q: The size of each q-batch.
+        batch_shape: The batch shape of the samples. If given, returns samples
+            of shape `n x *batch_shape x q x d`, where each batch is an
+            `n x q x d`-dim tensor of qMC samples.
         seed: The seed used for initializing Owen scrambling. If None (default),
             use a random seed.
 
     Returns:
-        A `n x q x d`-dim tensor of qMC samples from the box defined by bounds.
+        A `n [x *batch_shape] x q x d`-dim tensor of qMC samples from the box
+        defined by bounds.
 
     Example:
         >>> bounds = torch.stack([torch.zeros(3), torch.ones(3)])
         >>> samples = draw_sobol_samples(bounds, 10, 2)
     """
+    batch_shape = torch.Size([]) if batch_shape is None else batch_shape
+    batch_size = int(torch.prod(torch.tensor(batch_shape)))
     d = bounds.shape[-1]
     lower = bounds[0]
     rng = bounds[1] - bounds[0]
     sobol_engine = SobolEngine(q * d, scramble=True, seed=seed)
-    samples_raw = sobol_engine.draw(n, dtype=lower.dtype).view(n, q, d)
-    samples_raw = samples_raw.to(device=lower.device)
+    samples_raw = sobol_engine.draw(batch_size * n, dtype=lower.dtype)
+    samples_raw = samples_raw.view(*batch_shape, n, q, d).to(device=lower.device)
+    if batch_shape != torch.Size([]):
```

Comment:
if you explicitly check `batch_shape is None` above, then batch_shape here may be a tuple or a list, in which case the check fails and this breaks. Since empty tuples/lists evaluate to False you could do `batch_shape = batch_shape or torch.Size()`